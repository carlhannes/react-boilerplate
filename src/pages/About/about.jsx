import React from 'react';
import PropTypes from 'prop-types';

import Layout from '../../layout';

import Sidebar from '../../components/Sidebar';

function Counter({ count }) {
  return (
    <p>
      Counted to
      {' '}
      {count}
    </p>
  );
}

Counter.propTypes = {
  count: PropTypes.number.isRequired,
};

class About extends React.Component {
  state = {
    count: 0,
  }

  componentDidMount() {
    this.setState({ count: 10 });
  }

  addCounter = () => {
    this.setState(prevState => ({
      count: prevState.count + 1,
    }));
  }

  render() {
    const { count } = this.state;

    return (
      <Layout sidebar={(<Sidebar />)}>
        <h1>About</h1>
        <Counter count={count} />
        <button type="button" onClick={this.addCounter}>
          Click here to add to the counter!
        </button>
      </Layout>
    );
  }
}

export default About;
