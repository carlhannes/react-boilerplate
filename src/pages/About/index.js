import Page from './about';

export default Page;
