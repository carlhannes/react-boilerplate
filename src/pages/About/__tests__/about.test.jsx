import React from 'react';
import { mount } from 'enzyme';

import About from '../about';

jest.mock('../../../components/Sidebar/sidebar');

describe('About page', () => {
  it('should render correctly', () => {
    const wrapper = mount(<About />);
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });
});
