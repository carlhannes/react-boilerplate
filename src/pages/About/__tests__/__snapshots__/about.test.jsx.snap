// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page should render correctly 1`] = `
<About>
  <Layout
    sidebar={<Sidebar />}
  >
    <div
      className="sidebar"
    >
      <Sidebar>
        <ul>
          <li>
            sidebar mock
          </li>
        </ul>
      </Sidebar>
    </div>
    <main>
      <h1>
        About
      </h1>
      <Counter
        count={10}
      >
        <p>
          Counted to
           
          10
        </p>
      </Counter>
      <button
        onClick={[Function]}
        type="button"
      >
        Click here to add to the counter!
      </button>
    </main>
  </Layout>
</About>
`;
